tests/MyGame/Example/Monster.cs

```diff
@@ -441,39 +444,100 @@ public struct Monster : IFlatbufferObject
       for (var _j = _o.Test4.Count - 1; _j >= 0; --_j) { MyGame.Example.Test.Pack(builder, _o.Test4[_j]); }
       _test4 = builder.EndVector();
     }
-    var _testarrayofstring = _o.Testarrayofstring == null ? default(VectorOffset) : CreateTestarrayofstringVector(builder, _o.Testarrayofstring.Select(builder.CreateSharedString).ToArray());
-    var _testarrayoftables = _o.Testarrayoftables == null ? default(VectorOffset) : CreateTestarrayoftablesVector(builder, _o.Testarrayoftables.Select(__o => MyGame.Example.Monster.Pack(builder, __o)).ToArray());
+    var _testarrayofstring = default(VectorOffset);
+    if (_o.Testarrayofstring != null) {
+      var __testarrayofstring = new StringOffset[_o.Testarrayofstring.Count];
+      for (var _j = 0; _j < __testarrayofstring.Length; ++_j) { __testarrayofstring[_j] = builder.CreateSharedString(_o.Testarrayofstring[_j]); }
```

Comment:
Why does this default to `CreateSharedString` ? For data that is not repetitive, this is a lot of overhead in both memory and time.